### PR TITLE
[CRIMAPP-1099] only submit relevant employed information

### PR DESCRIPTION
**Pull request by timpeat**
## Description of change

Only show and submit required employment information. Handles:
-  changes in a users employment status
- changes in the partner's involvement
- changes in full means assessment status

## Link to relevant ticket

https://dsdmoj.atlassian.net/browse/CRIMAPP-1099

### CODE REVIEW

File: app/lib/errors.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module Errors
   class InvalidSession < StandardError; end
   class ApplicationCannotReceivePse < StandardError; end
+  class DateOfBirthPending < StandardError; end
+  class ApplicationCannotReceivePse < StandardError; end
```
Comment (ripan) [inline app/lib/errors.rb:5]: @timpeat Duplicate line

**Review by ripan — COMMENTED** [review]
Looks good to me

**Review by ripan — APPROVED** [review]

**Comment by arthurashman** [comment]
@timpeat I'm seeing Other Work Benefits on the final CYA page when I shouldn't

Route:
Client = employed
Answer all qs (inc other work bens = yes, add a number)
Reach final CYA
Return to Client employment status q
Change from employed to not employed
Return to final CYA - expect to not see other work bens section

**Comment by arthurashman** [comment]
Also switching from not working having added a bunch of means info, to passported, I get this error:
image

The code deletes the income object, then tries to look for employements

**Comment by arthurashman** [comment]
Second issue is not you, it's like that on a straight passported application first run through
I'll fix

**Comment by timpeat** [comment]
> @timpeat I'm seeing Other Work Benefits on the final CYA page when I shouldn't
> 
> Route: Client = employed Answer all qs (inc other work bens = yes, add a number) Reach final CYA Return to Client employment status q Change from employed to not employed Return to final CYA - expect to not see other work bens section

I'd missed off work benefits entirely. Will update.

**Review by arthurashman — APPROVED** [review]
Nice work! Great refactoring in there 🎉

**Pull request merged**